test/server/drain_manager_impl_test.cc

```diff
@@ -37,12 +40,15 @@ TEST(DrainManagerImplTest, All) {
   // Test drain sequence.
   Event::MockTimer* drain_timer = new Event::MockTimer(&server.dispatcher_);
```

Comment:
If you would like to me document this somewhere in the style guide let me know. It allows us to use unique_ptr in product code whenever possible at the expense of making some of the memory management in the tests a bit strange, which IMO is a good compromise.